include/mcl_3dl/pf.h

```diff
@@ -274,42 +274,57 @@ class ParticleFilter
     }
     return mean.getMean();
   }
-  std::vector<T> covariance(const FLT_TYPE pass_ratio = 1.0)
+  std::vector<T> covariance(
+      const FLT_TYPE pass_ratio = 1.0,
+      const FLT_TYPE random_sample_ratio = 1.0)
   {
     T e = expectation(pass_ratio);
     FLT_TYPE p_sum = 0;
-    size_t p_num = 0;
     std::vector<T> cov;
     cov.resize(e.size());
 
+    size_t p_num = 0;
     for (auto& p : particles_)
     {
       p_num++;
       p_sum += p.probability_;
       if (p_sum > pass_ratio)
         break;
     }
-    p_sum = 0;
-    for (auto& p : particles_)
+
+    const size_t num =
```

Comment:
The variable name "num" is too vague.
In addition, because it is used only in if-block in L304, It is better to move this within that if-block.

Reply:
addressed :+1: